src/strands_tools/base/tool_provider.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC
+
+from strands.tools.decorator import DecoratedFunctionTool
```

Comment:
This actually probably indicates that `DecoratedFunctionTool` needs to be exposed from `strands.tools`, but honestly I'd like to keep it as an implementation detail